netsync/manager.go

```diff
@@ -191,9 +199,26 @@ func (sm *SyncManager) startSync() {
 
 		syncPeer.PushGetBlockLocatorMsg(syncPeer.SelectedTip(), sm.dagParams.GenesisHash)
 		sm.syncPeer = syncPeer
-	} else {
-		log.Warnf("No sync peer candidates available")
+		return
+	}
+
+	log.Warnf("No sync peer candidates available")
+	if sm.dag.AdjustedTime().Sub(sm.dag.CalcPastMedianTime()) > time.Minute {
+		for peer, state := range sm.peerStates {
+			if !state.syncCandidate {
```

Comment:
Can't he become a sync candidate once you asked for it's tips?

Reply:
you need to know its selected tip before you start syncing from it